analytics/app/src/main/java/com/google/firebase/example/analytics/MainActivity.java

```diff
@@ -32,17 +33,32 @@ public class MainActivity extends AppCompatActivity
 // importing libraries to support 3rd party ad_impression snippets
         implements MaxAdRevenueListener, ImpressionDataListener {
 
+    // [START declare_analytics]
+    private FirebaseAnalytics mFirebaseAnalytics;
+    // [END declare_analytics]
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        // [START shared_app_measurement]
+        // Obtain the FirebaseAnalytics instance.
+        mFirebaseAnalytics = FirebaseAnalytics.getInstance(this);
+        // [END shared_app_measurement]
+
         enhancedEcommerce();
+
+        setUserFavoriteFood("avocado");
+
+        recordImageView();
+
+        recordScreenView();
```

Comment:
[The documentation](https://firebase.google.com/docs/analytics/screenviews#manually_track_screens) recommends logging in the `onResume` method. Which one is more appropriate?